code/game/machinery/machinery.dm

```diff
@@ -381,7 +382,7 @@ Class Procs:
 	return TRUE
 
 /obj/machinery/proc/is_in_range(var/mob/user)
-	if((!in_range(src, usr) || !istype(src.loc, /turf)) && !istype(usr, /mob/living/silicon))
+	if(((!in_range(src, usr) || !istype(src.loc, /turf)) && !istype(usr, /mob/living/silicon)) || !istype(usr, /mob/living/simple_animal/hostile/pulse_demon))
```

Comment:
I think this is wrong. I think this will allow a non-pulse demon to be in range of a machine at any distance, so long as they are not a pulse demon

Try using an air alarm, APC, or autolathe and walking away from it and manipulating it at a distance

Reply:
I didn't intend them to be "in range" since it'll area and APC checks are all I want.
Fixed the usr thing though.

Reply:
Actually after reading what you've said and thinking about it && is probably better.